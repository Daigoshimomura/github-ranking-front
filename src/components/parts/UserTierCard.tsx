import React from "react";
import { Card, CardMedia, CardContent, Typography } from "@material-ui/core";

interface UserTierCardProps {
  tierName: string;
  tierAbout: string;
  imageSrc: string;
  score: Number;
}

export class UserTierCard extends React.Component<UserTierCardProps, {}> {
  render() {
    return (
      <Card style={{ maxWidth: 270, display: "flex" }}>
        <CardMedia
          image={this.props.imageSrc}
          component="img"
          style={{ width: 100, margin: 15 }}
        ></CardMedia>
        <CardContent>
          <Typography color="primary" variant="h6" component="h6">
            {this.props.tierName}
          </Typography>
          <Typography color="textSecondary">
            Score : {this.props.score}
          </Typography>
          <Typography color="textSecondary">{this.props.tierAbout}</Typography>
        </CardContent>
      </Card>
    );
  }
}
