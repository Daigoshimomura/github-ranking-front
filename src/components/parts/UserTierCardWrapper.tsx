import React from "react";
import sample from "../../static/sample.png";
import {
  CHALLENGER,
  MASTER,
  DIAMOND,
  PRATINUM,
  GOLD,
  SILVER
} from "../../const/Tier";
import { UserTierCard } from "./UserTierCard";

interface UserTierCardWrapperProps {
  tier: string;
}

export class UserTierCardWrapper extends React.Component<
  UserTierCardWrapperProps,
  {}
> {
  render() {
    if (this.props.tier === CHALLENGER) {
      return (
        <UserTierCard
          tierName={this.props.tier}
          tierAbout="Top 5%"
          imageSrc={sample}
        />
      );
    } else if (this.props.tier === MASTER) {
      return (
        <UserTierCard
          tierName={this.props.tier}
          tierAbout="Top 10%"
          imageSrc={sample}
        />
      );
    } else if (this.props.tier === DIAMOND) {
      return (
        <UserTierCard
          tierName={this.props.tier}
          tierAbout="Top 15%"
          imageSrc={sample}
        />
      );
    } else if (this.props.tier === PRATINUM) {
      return (
        <UserTierCard
          tierName={this.props.tier}
          tierAbout="Top 20%"
          imageSrc={sample}
        />
      );
    } else if (this.props.tier === GOLD) {
      return (
        <UserTierCard
          tierName={this.props.tier}
          tierAbout="Top 30%"
          imageSrc={sample}
        />
      );
    } else if (this.props.tier === SILVER) {
      return (
        <UserTierCard
          tierName={this.props.tier}
          tierAbout="Top 40%"
          imageSrc={sample}
        />
      );
    } else {
      return (
        <UserTierCard
          tierName={this.props.tier}
          tierAbout="Under 40%"
          imageSrc={sample}
        />
      );
    }
  }
}
