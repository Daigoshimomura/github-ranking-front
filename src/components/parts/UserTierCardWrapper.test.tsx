import { mount } from "enzyme";
import React from "react";
import { UserTierCardWrapper } from "./UserTierCardWrapper";
import { UserTierCard } from "./UserTierCard";
import {
  CHALLENGER,
  MASTER,
  DIAMOND,
  PRATINUM,
  GOLD,
  SILVER,
  BRONZE
} from "../../const/Tier";
import bronze from "../../static/bronze.png";
import silver from "../../static/silver.png";
import gold from "../../static/gold.png";
import pratinum from "../../static/pratinum.png";
import diamond from "../../static/diamond.png";
import master from "../../static/master.png";
import challenger from "../../static/challenger.png";

describe("UserTierCardWrapper", () => {
  it("レンダリングの確認", () => {
    const wrapper = mount(<UserTierCardWrapper tier={"tier"} score={0} />);
    expect(wrapper.find(UserTierCard).length).toBe(1);
  });
  it("propsをちゃんと渡せてることの確認", () => {
    const wrapper = mount(<UserTierCardWrapper tier={CHALLENGER} score={0} />);
    const card = wrapper.find(UserTierCard);
    expect(card.prop("score")).toBe(0);
  });
  it("Challengerの場合", () => {
    const wrapper = mount(<UserTierCardWrapper tier={CHALLENGER} score={0} />);
    const card = wrapper.find(UserTierCard);
    expect(card.prop("tierAbout")).toBe("Top 5%");
    expect(card.prop("imageSrc")).toBe(challenger);
  });
  it("Masterの場合", () => {
    const wrapper = mount(<UserTierCardWrapper tier={MASTER} score={0} />);
    const card = wrapper.find(UserTierCard);
    expect(card.prop("tierAbout")).toBe("Top 10%");
    expect(card.prop("imageSrc")).toBe(master);
  });
  it("Diamondの場合", () => {
    const wrapper = mount(<UserTierCardWrapper tier={DIAMOND} score={0} />);
    const card = wrapper.find(UserTierCard);
    expect(card.prop("tierAbout")).toBe("Top 15%");
    expect(card.prop("imageSrc")).toBe(diamond);
  });
  it("Pratinumの場合", () => {
    const wrapper = mount(<UserTierCardWrapper tier={PRATINUM} score={0} />);
    const card = wrapper.find(UserTierCard);
    expect(card.prop("tierAbout")).toBe("Top 20%");
    expect(card.prop("imageSrc")).toBe(pratinum);
  });
  it("Goldの場合", () => {
    const wrapper = mount(<UserTierCardWrapper tier={GOLD} score={0} />);
    const card = wrapper.find(UserTierCard);
    expect(card.prop("tierAbout")).toBe("Top 30%");
    expect(card.prop("imageSrc")).toBe(gold);
  });
  it("Silverの場合", () => {
    const wrapper = mount(<UserTierCardWrapper tier={SILVER} score={0} />);
    const card = wrapper.find(UserTierCard);
    expect(card.prop("tierAbout")).toBe("Top 40%");
    expect(card.prop("imageSrc")).toBe(silver);
  });
  it("それ以外の場合", () => {
    const wrapper = mount(<UserTierCardWrapper tier={BRONZE} score={0} />);
    const card = wrapper.find(UserTierCard);
    expect(card.prop("tierAbout")).toBe("Under 40%");
    expect(card.prop("imageSrc")).toBe(bronze);
  });
});
