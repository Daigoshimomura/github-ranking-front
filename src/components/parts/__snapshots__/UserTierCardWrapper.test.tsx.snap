// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTierCardWrapper snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  style={
    Object {
      "display": "flex",
      "maxWidth": 270,
    }
  }
>
  <img
    className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
    src="bronze.png"
    style={
      Object {
        "margin": 15,
        "width": 100,
      }
    }
  />
  <div
    className="MuiCardContent-root"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary"
    >
      tier
    </h6>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      Score : 
      0
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      Under 40%
    </p>
  </div>
</div>
`;
